### PR TITLE
Update docs/docstrings

### LINKED ISSUE #2194: `regrid!` is (a bit) more general than it advertises
After #2186 I believe that the warning in the docstring of `regrid!` should be amended.

https://github.com/CliMA/Oceananigans.jl/blob/e90b22bcfcf3cba53ed61e46024d3a759f35ef56/src/Fields/regridding_fields.jl#L9-L16

### PR DESCRIPTION
**Pull request by navidcy**
A few updates.

Closes #2194.

### CODE REVIEW

**Review by glwagner — APPROVED** [review]

File: src/Fields/field_reductions.jl [hunk 1]
```diff
@@ -13,21 +13,35 @@ end
 
 Return a `Reduction` of `operand` with `reduce!`, along `dims`.
 
-elements for which `condition(i ,j ,k) == true` are masked with a `mask`
+Field elements for which `condition(i ,j ,k) == true` are masked with a `mask`.
 
 Example
```
Comment (tomchor) [inline src/Fields/field_reductions.jl:18]: Should we mention somewhere here that the reduction must happen in-place? (Or is that not the case?)

Comment (navidcy) [reply src/Fields/field_reductions.jl:18]: you mean that the reduction operator must be a function that works 'in place'?

Comment (tomchor) [reply src/Fields/field_reductions.jl:18]: Yes!

Comment (navidcy) [reply src/Fields/field_reductions.jl]: Done via  9713191

**Review by tomchor — APPROVED** [review]

**Review by simone-silvestri — APPROVED** [review]

**Pull request merged**